control/tests/sisotool_test.py

```diff
@@ -140,3 +140,25 @@ def test_sisotool_mimo(self,  sys222, sys221):
         # but 2 input, 1 output should
         with pytest.raises(ControlMIMONotImplemented):
             sisotool(sys221)
+
+@pytest.mark.usefixtures("mplcleanup")
+class TestPidDesigner:
+    syscont = TransferFunction(1,[1, 3, 0])
+    sysdisc1 = c2d(TransferFunction(1,[1, 3, 0]), .1)
+    syscont221 = StateSpace([[-.3, 0],[1,0]],[[-1,],[.1,]], [0, -.3], 0)
```

Comment:
I think the current CI failure is due to this code.

You have to hide the system creation in a fixture. Calling those constructors in the class definition means constructing the systems during test collection, where the matrix/array switch is not set up yet.

Reply:
I can't seem to get it to work. When I do eg. this: 

``` python
class TestPidDesigner:
   @pytest.fixture
    def syscont(self):
        return TransferFunction(1,[1, 3, 0])

    @pytest.fixture
    def sysdisc1(self):
        return c2d(TransferFunction(1,[1, 3, 0]), .1)

    # cont or discrete, vary P I or D
    @pytest.mark.parametrize('plant', (syscont, sysdisc1))
    @pytest.mark.parametrize('gain', ('P', 'I', 'D'))
    @pytest.mark.parametrize("kwargs", [{'Kp0':0.01},])
    def test_pid_designer_1(self, plant, gain, kwargs):
        pid_designer(plant, gain, **kwargs)```
The `pid_designer` function does not get the actual transfer function, it gets the fixture instead. 
This also raises an error because 'plant' is not found: `    @pytest.mark.parametrize('plant', ('syscont', 'syscont221'), indirect=True)`
Any idea how I give it the actual system?

Reply:
Like `tsystem` in `timeresp_test.py`:

```python
class TestPidDesigner:
    @pytest.fixture
    def plant(self, request):
        plants = {
            'syscont': TransferFunction(1,[1, 3, 0]),
            'sysdisc1': c2d(TransferFunction(1,[1, 3, 0]), .1)
        }
        return plants[request.param]

    # cont or discrete, vary P I or D
    @pytest.mark.parametrize('plant', ('syscont', 'sysdisc1'), indirect=True)
    @pytest.mark.parametrize('gain', ('P', 'I', 'D'))
    @pytest.mark.parametrize("kwargs", [{'Kp0':0.01},])
    def test_pid_designer_1(self, plant, gain, kwargs):
        pid_designer(plant, gain, **kwargs)

Reply:
That did it, thanks.